File: user_mobile/lib/models/task.dart
class Task {
  final int id;
  final String title;
  final String description;
  final int adminId;
  final int? assignedTo;
  final String status; // pending, accepted, completed, rejected
  final DateTime createdAt;
  final DateTime? acceptedAt;
  final DateTime? completedAt;
  final String? adminName;
  final String? assignedUserName;

  Task({
    required this.id,
    required this.title,
    required this.description,
    required this.adminId,
    this.assignedTo,
    required this.status,
    required this.createdAt,
    this.acceptedAt,
    this.completedAt,
    this.adminName,
    this.assignedUserName,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      adminId: json['admin_id'],
      assignedTo: json['assigned_to'],
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      acceptedAt: json['accepted_at'] != null
          ? DateTime.parse(json['accepted_at'])
          : null,
      completedAt: json['completed_at'] != null
          ? DateTime.parse(json['completed_at'])
          : null,
      adminName: json['admin']?['name'],
      assignedUserName: json['assigned_user']?['name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'admin_id': adminId,
      'assigned_to': assignedTo,
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'accepted_at': acceptedAt?.toIso8601String(),
      'completed_at': completedAt?.toIso8601String(),
    };
  }
}

File: admin-web/app/Models/Task.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Task extends Model
{
    protected $fillable = [
        'title',
        'description',
        'admin_id',
        'assigned_to',
        'status',
        'offered_at',
        'accepted_at',
        'rejected_at',
        'completed_at',
    ];

    protected $casts = [
        'offered_at' => 'datetime',
        'accepted_at' => 'datetime',
        'rejected_at' => 'datetime',
        'completed_at' => 'datetime',
    ];

    public function admin(): BelongsTo
    {
        return $this->belongsTo(User::class, 'admin_id');
    }

    public function assignedUser(): BelongsTo
    {
        return $this->belongsTo(User::class, 'assigned_to');
    }
}
